src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -583,6 +593,7 @@ object AppDefinition extends GeneralPurposeCombinators {
     appDef must complyWithSingleInstanceLabelRules
     appDef must complyWithReadinessCheckRules
     appDef must complyWithUpgradeStrategyRules
+    appDef must complyWithGpuRules
```

Comment:
If the default containerizer is Mesos, than we should probably remove `app.container.isDefined` to only `app.container.exists(_.docker.isEmpty)`